### PR TITLE
v1.14 Backports 2024-06-25

### PR DESCRIPTION
**Pull request by YutaroHayakawa**
* [ ] #26779 (@ferozsalam)
 * [x] #32951 (@giorio94) :warning: resolved conflicts
 * [x] #33216 (@aanm)
 * [ ] #33218 (@dependabot[bot]) :warning: resolved conflicts
 * [x] #33292 (@joestringer)
 * [x] #33353 (@aanm)
 * [ ] #33304 (@learnitall)
 * [x] #33368 (@aanm)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 26779 32951 33216 33218 33292 33353 33304 33368
```

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -382,8 +382,13 @@ func FormatStatusResponse(w io.Writer, sr *models.StatusResponse, sd StatusDetai
 	}
 
 	if sr.ClusterMesh != nil {
+<<<<<<< HEAD
 		fmt.Fprintf(w, "ClusterMesh:\t%d/%d clusters ready, %d global-services\n",
 			numReadyClusters(sr.ClusterMesh), len(sr.ClusterMesh.Clusters), sr.ClusterMesh.NumGlobalServices)
+=======
+		fmt.Fprintf(w, "ClusterMesh:\t%d/%d remote clusters ready, %d global-services\n",
+			NumReadyClusters(sr.ClusterMesh.Clusters), len(sr.ClusterMesh.Clusters), sr.ClusterMesh.NumGlobalServices)
+>>>>>>> 2c68c61947 (cilium-dbg: minor clarifications to the clustermesh status output)
```
Comment (giorio94) [inline pkg/client/client.go]: Something went wrong in the conflict resolution here. Should additionally change `failures` to `reconnections` a couple of lines below.

**Review by giorio94 — CHANGES_REQUESTED** [review]
Thanks, and sorry for the conflicts! Just a comment inline about an incomplete conflict resolution.

File: pkg/client/client.go (hunk 1 above)
Comment (YutaroHayakawa) [reply pkg/client/client.go]: Should be fixed now.

**Review by ferozsalam — APPROVED** [review]

**Review by giorio94 — APPROVED** [review]
My commits look good, thanks!

**Review by aanm — APPROVED** [review]

**Comment by YutaroHayakawa** [comment]
/test-backport-1.14

**Review by joestringer — APPROVED** [review]
Approving for @cilium/github-sec and my PR.

**Review by learnitall — APPROVED** [review]
Thanks!

**Review by viktor-kurchenko — APPROVED** [review]

**Review by nathanjsweet — APPROVED** [review]

**Review by youngnick — APPROVED** [review]

**Comment by YutaroHayakawa** [comment]
The failing `Image CI Build / Display Digests` should be fixed by https://github.com/cilium/cilium/pull/33216 included in this PR.

**Pull request merged**